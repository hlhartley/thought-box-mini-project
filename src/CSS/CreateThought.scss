// .create-thought-container {
//     display: flex;
//     flex-flow: column;
//     text-align: center;
// }

#title-input {
    height: 21px;
    width: 500px;
}

#body-input {
    height: 140px;
    width: 500px;
}

.submit-btn {
    margin: 10px 0 20px 0;
    height: 35px;
    color: white;
    background-color: grey;
    width: 518px;
}