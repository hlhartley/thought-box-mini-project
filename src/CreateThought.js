import React, { Component } from 'react';

class CreateThought extends Component {
  constructor() {
    super();

  this.state = {
    title: '',
    body: ''
  }
}

  handleChange(e) {
    const { name, value } = e.target
    this.setState({[name]: value})
  }
  
  handleSubmit() {
    e.preventDefault();
    this.props.createThought(this.state);
    this.setState({title: '', body: ''});
    this.clearInputFields();
  }
  
  clearInputFields() {
    document.getElementById('title-input').value = '';
    document.getElementById('body-input').value = '';
  }

  render() {
    const { title, body } = this.state
    return(
      <form className='create-thought-container' onSubmit={this.handleSubmit.bind(this)}>
        Title: <input value={title} type='text' name='title' id='title-input' onChange={this.handleChange.bind(this)}></input>
        Body: <input value={body} type='text' name='body' id='body-input' onChange={this.handleChange.bind(this)}></input>
        <button name='submit' className='submit-btn'>Submit</button>
      </form>
    );
  }
}

export default CreateThought
