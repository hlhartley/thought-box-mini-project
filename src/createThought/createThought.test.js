import React from 'react';
import { shallow, mount } from 'enzyme';

import CreateThought from './CreateThought.js';

describe('CreateThought', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<CreateThought />);
    expect(wrapper).toMatchSnapshot()
  });

  it('updates the state of the title field', () => {
    const wrapper = mount(<CreateThought/>);
    const mockEvent = { target: { value: 'abc', name: 'title' } }
    const expectedState = {
      title: 'abc',
      body: ''
    };
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state()).toEqual(expectedState);
  });

  it('updates the state of the body field', () => {
    const wrapper = mount(<CreateThought/>);
    const mockEvent = { target: { value: 'abc', name: 'body' } }
    const expectedState = {
      title: '',
      body: 'abc'
    };
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state()).toEqual(expectedState);
  });

  it('calls createThought prop function with the data from state as an argument, and input fields go back to empty strings', () => {
    const createThoughtMock = jest.fn();
    const wrapper = shallow(
      <CreateThought createThought={createThoughtMock} />
    );
    const expectedState = {
      title: '',
      body: ''
    };
    wrapper.setState({title: 'no', body: 'one'})

    // How do we call handleSubmit?
    wrapper.instance().handleSubmit()
    
    // How do we assert that our mock was called with the
    // correct params?
    expect(createThoughtMock).toHaveBeenCalledWith({title: 'no', body: 'one'});
    wrapper.find('.submit-btn').simulate('click')
    expect(wrapper.state()).toEqual(expectedState);
  });

});
