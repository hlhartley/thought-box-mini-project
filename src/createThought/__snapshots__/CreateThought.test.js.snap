// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateThought should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateThought />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "CreateThought",
        <input
          className="title"
          name="title"
          onChange={[Function]}
          type="text"
        />,
        <input
          className="body"
          name="body"
          onChange={[Function]}
          type="text"
        />,
        <input
          className="submit-btn"
          name="submit"
          onSubmit={[Function]}
          type="submit"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      "CreateThought",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "title",
          "name": "title",
          "onChange": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "body",
          "name": "body",
          "onChange": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "submit-btn",
          "name": "submit",
          "onSubmit": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "CreateThought",
          <input
            className="title"
            name="title"
            onChange={[Function]}
            type="text"
          />,
          <input
            className="body"
            name="body"
            onChange={[Function]}
            type="text"
          />,
          <input
            className="submit-btn"
            name="submit"
            onSubmit={[Function]}
            type="submit"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "CreateThought",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "title",
            "name": "title",
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "body",
            "name": "body",
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "submit-btn",
            "name": "submit",
            "onSubmit": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
